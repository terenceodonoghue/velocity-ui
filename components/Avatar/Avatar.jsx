/** @jsx jsx */
import { jsx } from '@emotion/core';
import * as S from './Avatar.styles';

const Round = props => <S.Round {...props} />;
const Square = props => <S.Square {...props} />;

export default {
  Round,
  Square,
};
