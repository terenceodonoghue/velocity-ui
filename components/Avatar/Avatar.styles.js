/* eslint-disable import/prefer-default-export */
import styled from '@emotion/styled';

export const Round = styled.img`
  border-radius: 50%;
`;
