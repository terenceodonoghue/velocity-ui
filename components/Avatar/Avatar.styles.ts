import { css, InterpolationWithTheme } from '@emotion/core';
import { Sizable, Theme } from '~/types';

export const round = ({
  size,
}: Sizable): InterpolationWithTheme<Theme> => () => css`
  border-radius: 50%;
  height: ${size}px;
  width: ${size}px;
`;

export const square = ({
  size,
}: Sizable): InterpolationWithTheme<Theme> => () => css`
  border-radius: 6px;
  height: ${size}px;
  width: ${size}px;
`;
