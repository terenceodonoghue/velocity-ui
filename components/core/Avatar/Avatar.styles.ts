import { css } from '@emotion/react';

export const rounded = css`
  border-radius: 50%;
`;

export const square = css`
  border-radius: 6px;
`;
