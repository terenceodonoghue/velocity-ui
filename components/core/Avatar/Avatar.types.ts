import { ImgHTMLAttributes } from 'react';

export interface AvatarProps extends ImgHTMLAttributes<HTMLImageElement> {
  alt: string;
  size?: number;
  variant?: 'rounded' | 'square';
}
