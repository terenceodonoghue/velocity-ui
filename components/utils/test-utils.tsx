import React, { FunctionComponent, ReactElement } from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import ThemeProvider from '~/components/core/ThemeProvider';

const Providers: FunctionComponent = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

export { customRender as render };
