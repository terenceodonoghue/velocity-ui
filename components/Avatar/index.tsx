/** @jsx jsx */
import { jsx } from '@emotion/core';
import { FunctionComponent } from 'react';
import { AvatarProps } from '~/types';
import * as css from './Avatar.styles';

const Avatar: FunctionComponent<AvatarProps> = ({
  alt,
  round,
  size,
  ...props
}) => (
  <img
    alt={alt}
    css={round ? css.round({ size }) : css.square({ size })}
    {...props}
  />
);

export default Avatar;
