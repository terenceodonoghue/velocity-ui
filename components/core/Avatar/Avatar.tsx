import React, { FunctionComponent } from 'react';
import * as css from './Avatar.styles';
import type { AvatarProps } from './Avatar.types';

const Avatar: FunctionComponent<AvatarProps> = ({
  alt,
  size,
  variant = 'rounded',
  ...props
}) => (
  <img
    alt={alt}
    css={[
      variant === 'rounded' ? css.rounded : css.square,
      { height: size, width: size },
    ]}
    data-testid="avatar"
    {...props}
  />
);

export default Avatar;
