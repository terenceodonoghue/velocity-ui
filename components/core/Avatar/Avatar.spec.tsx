import React from 'react';
import faker from 'faker';
import { render } from 'test-utils';
import Avatar from './Avatar';

describe('Avatar', () => {
  it('is rounded by default', () => {
    const { container, getByTestId } = render(
      <Avatar alt={faker.name.findName()} src={faker.image.avatar()} />,
    );

    expect(getByTestId('avatar')).toHaveStyle({ borderRadius: '50%' });
    expect(container).toMatchSnapshot();
  });

  it('renders rounded variant', () => {
    const { container, getByTestId } = render(
      <Avatar
        alt={faker.name.findName()}
        src={faker.image.avatar()}
        variant="rounded"
      />,
    );

    expect(getByTestId('avatar')).toHaveStyle({ borderRadius: '50%' });
    expect(container).toMatchSnapshot();
  });

  it('renders square variant', () => {
    const { container, getByTestId } = render(
      <Avatar
        alt={faker.name.findName()}
        src={faker.image.avatar()}
        variant="square"
      />,
    );

    expect(getByTestId('avatar')).toHaveStyle({ borderRadius: '6px' });
    expect(container).toMatchSnapshot();
  });
});
