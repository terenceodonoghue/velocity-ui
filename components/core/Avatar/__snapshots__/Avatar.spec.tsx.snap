// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar is rounded by default 1`] = `
.emotion-0 {
  border-radius: 50%;
}

<div>
  <img
    alt="Clara Keebler"
    class="emotion-0"
    data-testid="avatar"
    src="https://s3.amazonaws.com/uifaces/faces/twitter/giancarlon/128.jpg"
  />
</div>
`;

exports[`Avatar renders rounded variant 1`] = `
.emotion-0 {
  border-radius: 50%;
}

<div>
  <img
    alt="Beulah Reichel"
    class="emotion-0"
    data-testid="avatar"
    src="https://s3.amazonaws.com/uifaces/faces/twitter/uxpiper/128.jpg"
  />
</div>
`;

exports[`Avatar renders square variant 1`] = `
.emotion-0 {
  border-radius: 6px;
}

<div>
  <img
    alt="Elisa Jenkins"
    class="emotion-0"
    data-testid="avatar"
    src="https://s3.amazonaws.com/uifaces/faces/twitter/betraydan/128.jpg"
  />
</div>
`;
